src/libsyntax/parse/token.rs

```diff
@@ -201,6 +201,7 @@ impl Token {
             OpenDelim(..)               => true, // tuple, array or block
             Literal(..)                 => true, // literal
             Not                         => true, // operator not
+            Tilde                       => true, // operator not (incorrect)
```

Comment:
Yeah, I agree. I have deleted this line, and the test works fine. At least for this case, it is not necessary  to modify `can_begin_expr` .